### ISSUE TITLE
Initial plans for pyhf-validation

### ISSUE THREAD

**Issue by matthewfeickert** [pyhf/pyhf-validation#1]
Transferring some information and ideas from a developer chat with @kratsg:

- [ ] Support CLI that will allow for 
```
pyhf-validation spec.json
```
that will run `pyhf cls` over `spec.json` _and_ additionally runs `json2xml`+`hist2workspace`+`run_single.py` on the resulting ROOT workspace from the HistFactory side.
- [ ] Provide signatures of
```python
delta, pyhf_out, hifa_out = pyhf-validation.run_validation("spec.json")
```
- [ ] Support ROOT and PyROOT in automated testing

**Comment by matthewfeickert** [comment]
@danikam has already done a lot of really nice work (c.f. [SUSY_1Lbb_xchecks](https://gitlab.cern.ch/damacdon/susy_1lbb_xchecks)), so it will be important to take what parts of code she already has and port them over when possible.

**Comment by danikam** [comment]
> @danikam has already done a lot of really nice work (c.f. [SUSY_1Lbb_xchecks](https://gitlab.cern.ch/damacdon/susy_1lbb_xchecks)), so it will be important to take what parts of code she already has and port them over when possible.
^^with much help from @lukasheinrich + @kratsg :)

**Comment by danikam** [comment]
I can work on converting the 1Lbb checks to standalone python functions (assuming ROOT and pyhf dependencies present). Should these go into `src/hfval`?

**Comment by kratsg** [comment]
I was thinking perhaps you can put them in top-level `scripts/` and then slowly migrate into `src/hfval` ?

**Comment by danikam** [comment]
Ok, will do!

**Comment by matthewfeickert** [comment]
Yeah if you can make a PR for the `scripts/` then we can do the migration together into `src/`
